explainability-core/src/main/java/org/kie/trustyai/explainability/model/DataframeMetadata.java

```diff
@@ -0,0 +1,190 @@
+package org.kie.trustyai.explainability.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.kie.trustyai.explainability.model.domain.EmptyFeatureDomain;
+import org.kie.trustyai.explainability.model.domain.FeatureDomain;
+
+public class DataframeMetadata {
+    private final List<String> names;
+    private List<String> nameAliases;
+    private List<String> cachedColumnNames;
+    private List<Type> types;
+    private final List<Boolean> constrained;
+    private List<FeatureDomain> domains;
+    private final List<Boolean> inputs;
+
+    // constructors ====================================================================================================
+    public DataframeMetadata() {
+        this(new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
+    }
+
+    public DataframeMetadata(List<String> names, List<String> nameAliases, List<Type> types, List<Boolean> constrained, List<FeatureDomain> domains,
+            List<Boolean> inputs) {
+        this.names = new ArrayList<>(names);
+        this.nameAliases = new ArrayList<>(nameAliases);
+        this.types = new ArrayList<>(types);
+        this.constrained = new ArrayList<>(constrained);
+        this.domains = new ArrayList<>(domains);
+        this.inputs = new ArrayList<>(inputs);
+        updateCachedColumnNames();
+    }
+
+    public DataframeMetadata copy() {
+        return new DataframeMetadata(
+                new ArrayList<>(this.names),
+                new ArrayList<>(this.nameAliases),
+                new ArrayList<>(this.types),
+                new ArrayList<>(this.constrained),
+                new ArrayList<>(this.domains),
+                new ArrayList<>(this.inputs));
+    }
+
+    public void remove(int column) {
+        names.remove(column);
+        nameAliases.remove(column);
+        types.remove(column);
+        constrained.remove(column);
+        domains.remove(column);
+        inputs.remove(column);
+        cachedColumnNames.remove(column);
+    }
+
+    // row adders ======================================================================================================
```

Comment:
would it be possible to unit test the _row adders_ and _column accessors_ behaviors?